spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.actuate.cassandra;
+
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.querybuilder.Select;
+import org.springframework.boot.actuate.health.AbstractReactiveHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.data.cassandra.ReactiveResultSet;
+import org.springframework.data.cassandra.core.ReactiveCassandraOperations;
+import org.springframework.util.Assert;
+import reactor.core.publisher.Mono;
+
+/**
+ * Simple implementation of a {@link HealthIndicator} returning status information for
+ * Cassandra data stores.
+ *
+ * @author Artsiom Yudovin
+ * @since 2.0.0
+ */
+public class CassandraReactiveHealthIndicator extends AbstractReactiveHealthIndicator {
+
+	private final ReactiveCassandraOperations reactiveCassandraOperations;
+
+	/**
+	 * Create a new {@link CassandraHealthIndicator} instance.
+	 * @param reactiveCassandraOperations the Cassandra operations
+	 */
+	public CassandraReactiveHealthIndicator(ReactiveCassandraOperations reactiveCassandraOperations) {
+		Assert.notNull(reactiveCassandraOperations, "ReactiveCassandraOperations must not be null");
+		this.reactiveCassandraOperations = reactiveCassandraOperations;
+	}
+
+	@Override
+	protected Mono<Health> doHealthCheck(Health.Builder builder) {
+		Select select = QueryBuilder.select("release_version").from("system", "local");
+		Mono<ReactiveResultSet> results = this.reactiveCassandraOperations.getReactiveCqlOperations()
```

Comment:
How about simplifying the CQL call to:

```java
Mono<String> version = template.getReactiveCqlOperations().queryForObject(select, String.class);
```
?

This requires less mocking in the tests. Would be also good to improve `CassandraHealthIndicator` to call `queryForObject(…)`.

Reply:
ok, I agree with you and have changed it, thx